UITests/UITests.Tests.Shared/Controls/TextBoxMaskTest.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Windows.Apps.Test.Foundation.Controls;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Common;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Infra;
+
+#if USING_TAEF
+using WEX.Logging.Interop;
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+#endif
+
+namespace UITests.Tests
+{
+    [TestClass]
+    public class TextBoxMaskTest : UITestBase
+    {
+        [ClassInitialize]
+        [TestProperty("RunAs", "User")]
+        [TestProperty("Classification", "ScenarioTestSuite")]
+        [TestProperty("Platform", "Any")]
+        public static void ClassInitialize(TestContext testContext)
+        {
+            TestEnvironment.Initialize(testContext, WinUICsUWPSampleApp);
+        }
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            TestEnvironment.AssemblyCleanupWorker(WinUICsUWPSampleApp);
+        }
+
+        [TestMethod]
+        [TestPage("TextBoxMaskTestPage")]
+        public void TestTextBoxMaskBinding_Property()
+        {
+            var initialValue = FindElement.ById<TextBlock>("InitialValueTextBlock").GetText();
+            var textBox = FindElement.ById<Edit>("TextBox");
+
+            Verify.AreEqual(initialValue, textBox.GetText());
+
+            var changeButton = FindElement.ById<Button>("ChangeButton");
```

Comment:
It's a real shame you can't do something like this here:

`var changeButton = FindElement.ById<Button>(nameof(UITests.App.PagesTextBoxMaskTestPage.ChangeButton));`

Magic strings are bad coz they can break. If the page being tested was changed, it would be great if the compiler could tell us the test was no longer valid...
All theoretical in this instance as it's just a sample, but still.


Aside: Do you know why the compiler can't resolve this in the MSTest app? Is it because the netcoreapp doesn't include (and compile) the XAML by default?

Reply:
@mrlacey yeah, we're doing some shared project magic to put the two pieces together, but they both end up being compiled in different programs. So the MSTestApp doesn't have the dependencies to know what the XAML Page is and understand what it contains.

We could theoretically use a source generator or something to take the pages and output interfaces to mimic them that could be used similarly though... 🤔 I have no idea how easy or hard something like that would be though...

Reply:
Had a look at using C#9SourceGenerators but can't access the XAML files :(
Would have been nice.

Reply:
@mrlacey no worries, thanks for taking a look! It was just a though, but good to know for the future.